stencils/pace/stencils/testing/temporaries.py

```diff
@@ -40,10 +40,9 @@ def _assert_same_temporaries(dict1: dict, dict2: dict) -> List[str]:
         attr2 = dict2[attr]
         if isinstance(attr1, np.ndarray):
             try:
-                np.testing.assert_almost_equal(
-                    attr1, attr2, err_msg=f"{attr} not equal"
-                )
-            except AssertionError:
+                assert np.allclose(attr1, attr2, equal_nan=True)
```

Comment:
Do we have NaNs in our temporaries? Is there another reason for this change?

Reply:
Yes. Grid/Metric terms generation makes NaNs

Reply:
I don't recall this happening in the fortran as the debug mode checks for all but underflows - which are ftz'd.  Do you know what causes this in the python code?

Reply:
We have NaN in the geometry under some circumstances at least:
```
/home/runner/work/pace/pace/util/pace/util/grid/geometry.py:516: RuntimeWarning: invalid value encountered in divide
    del6_v = sina_u * dy / dxc
```

I believe those are in the halo